projects/exui-common-lib/src/lib/services/idle/idle.services.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import {DocumentInterruptSource, Idle, } from '@ng-idle/core';
+import {Keepalive} from '@ng-idle/keepalive';
+import {Observable, Subject} from 'rxjs';
+import {
+  delay,
+  distinctUntilChanged,
+  map
+} from 'rxjs/operators';
+import {IdleConfigModel} from '../../models/idle-config.model';
+
+@Injectable()
+export class IdleService {
+  private appStateEmitter: Subject<{type: string, countdown?: string; isVisible?: boolean}>;
```

Comment:
Not sure if this output is the correct one for a library service, it seems too tied to the particular implementation of a logout modal. I'd expect this service to emit when different idle thresholds are triggered. It's up to the implementing app to determine what does then relate to (such as displaying a modal, or logging the user out, etc).

So I'd refactor this service in a way where it takes in a list of triggers (name and timeout). When the user has been idle for [timeout], then [name] is emitted as a trigger. That way it's flexible to be used for anything that needs to act on idle times